### PR TITLE
fix(i18n): fallback to English for missing or empty translations

### PR DESCRIPTION
**Pull request by Achiket123**
What type of PR is this?
bug(#864 )

What this PR does / why we need it:
Sometimes when the app talks in other languages, it doesn’t know what to say, so it just says nothing or leaves things blank. That makes people confused.

This PR helps it always say the English words if it doesn’t know how to say something in another language. So even if the app doesn’t understand everything, it can still talk clearly and the screen won’t look broken.

It makes the app easier to use when some parts are not fully translated yet.

Summary of changes:
Changed the i18n.ts file.

Added a thing (called postTranslation hook) that checks if the app is saying nothing, and then it tells it to say the English words or just show the key instead.

Also changed the missing handler to do the same thing if it can’t find a translation.

Used a helper called getNested to help find words inside deeper spots, like "components.buttons.save.label".

Which issue(s) this PR fixes:
Fixes #<issue-number-for-missing-translation-fallback>
(You can put the real issue number here.)

Special notes for your reviewer:
Please look at the part where the app decides what to say (the i18n plugin logic). Make sure it always shows English words when it can’t find anything, even if it’s just an empty string.

The translation files were not changed — only the logic was updated.

Testing
I changed the app language to something where lots of words are missing (like Tamil or Chinese (Hong Kong)).
I checked that if a word is not there or is just empty, the app shows the English word instead.
I also made sure it works for simple keys like "title" and also for deep keys like "screens.home.button.save".

### CODE REVIEW

File: frontend/plugins/i18n.ts [hunk 1]
```diff
@@ -38,6 +63,11 @@ export default defineNuxtPlugin(({ vueApp }) => {
   };
 });
 
+// Utility to resolve nested keys like 'components.app.create_modal.enter'
+function getNested(obj: any, path: string) {
+  return path.split('.').reduce((o, k) => (o && o[k] !== undefined ? o[k] : undefined), obj);
+}
```
[the remark this answers is not in the document]
Comment (Achiket123) [reply frontend/plugins/i18n.ts]: It does not always set to English. It does so only when it is unable to find the key or string is empty.

[the remark this answers is not in the document]
Comment (Achiket123) [reply frontend/plugins/i18n.ts]: Corrected the getNested as suggested by coderabbit

**Comment by tonyaellie** [comment]
Does it not already fallback to the English version?

**Comment by Achiket123** [comment]
> Does it not already fallback to the English version?

Nope , in the latest version it was not as the bug report stated and i also tested it, i selected Tamil and every text became empty string. So i fixed it.

**Comment by tankerkiller125** [comment]
I think what's happening is weblate includes the keys for untranslated strings in the generated output. And for some reason the library used is apparently not filtering out empty strings because I guess empty strings are valid translations?

A better way to fix this would be to find a way to make weblate not include untranslated strings in the final output.

Edit: I've updated weblate to remove empty strings from the translation output. This should be testable in https://dev.homebox.software in the next few hours.

**Comment by tankerkiller125** [comment]
Closing this PR, updating Weblate to ensure that translation keys with no translation are not committed to the repo seems to have resolved the issue in https://dev.homebox.software
